.github/workflows/merged-pr.yml

```diff
@@ -0,0 +1,23 @@
+name: Merged Pull Request
+
+# only trigger on pull request closed events
+on:
+  pull_request:
+    types: [ closed ]
+
+jobs:
+  merge_job:
+    # this job will only run if the PR has been merged
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v5
+        with:
+          github-token: ${{secrets.ELEVATED_GITHUB_TOKEN}}
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: "If this is a user-visible change, remember to update the changelog on the appropriate release branch."
```

Comment:
Can we link to instructions as to how to write an appropriate changelog message? Or is that just kinda obvious?

Reply:
Don't let this block the PR. Maybe for a future change once we see how this works.

Reply:
We're intending this message to be for the person who did the merge (someone who works at HashiCorp) rather than the person who submitted the PR, and so I think we're assuming they already know what they need to do and we're basically just creating a GitHub notification to remind them to do it.

Given that though, it might be worth making it clearer in the message text that its audience _is_ the maintainer who merged it, so that if we're tardy in doing the changelog update we don't make the original submitter feel like they must e.g. open _another_ PR to suggest a changelog addition, or whatever.

Reply:
> Can we link to instructions as to how to write an appropriate changelog message?

Good thought! I don't think we currently _have_ any such instructions and I don't currently have the capacity to try writing them. When they do exist, we should update this hook to point to them.

Reply:
Thanks! I was thinking the "how to write a changelog entry" would be targeted towards an internal contributor, although given my role I suppose I should be thinking more of our external audience. 😅